test/Dialect/XTenNN/ops_invalid.mlir

```diff
@@ -24,3 +24,32 @@ func.func @mish_int(%arg0: tensor<1x10xi4>) -> tensor<1x10xi4> {
     %0 = xten_nn.mish %arg0 : (tensor<1x10xi4>) -> tensor<1x10xi4>
     return %0 : tensor<1x10xi4>
 }
+
+// -----
+
+func.func @kernel_missing_parenthesis() {
+    // expected-error@+1 {{expected '('}}
+    %a = xten_nn.kernel "myKernel" -> tensor<2xi64>
+}
+
+// -----
+
+func.func @kernel_missing_type(%arg0: i8, %arg1: i8) {
+    // expected-error@+1 {{expected ':'}}
+    %a = xten_nn.kernel "myKernel" (%arg0, %arg1) -> tensor<2xi64>
+}
+
+// -----
+
+func.func @kernel_trailing_comma(%arg0: i8) {
+    // expected-error@+1 {{expected SSA operand}}
+    %a = xten_nn.kernel "myKernel" (%arg0 :i8, ) -> tensor<2xi64>
+}
+
+// -----
+
+func.func @kernel_missing_name() {
+    // expected-error@+1 {{custom op 'xten_nn.kernel' invalid kind of attribute specified}}
```

Comment:
Error message here could also be more meaningful, we know we are missing the `name` attribute.

Reply:
Unfortunately, having better errors here seems to require to not use any of the parser convenience methods and reimplement most of them. I don't think its worth the effort.